3d/splitflap.scad

```diff
@@ -361,10 +362,32 @@ module flap() {
     }
 }
 
-module translated_flap() {
-    translate([0, flap_pitch_radius, 0]) {
-        rotate([flap_rendered_angle, 0, 0]) {
-            flap();
+module draw_letter(letter) {
+    color([0,0,0])
+    linear_extrude(height=0.1, center=true)
+        translate([0, -flap_height * letter_height, 0])  // valign compensation
+            scale([letter_width, 1, 1])
+                translate([letter_offset_x, letter_offset_y])
+                    text(text=letter, size=flap_height * letter_height * 2, font=letter_font, halign="center");
+}
+
+module flap_letter(letter, half = 0) {
+    translate([0, 0, flap_thickness/2 + eps]) {
+        if(half != 0) {  // trimming to top (1) or bottom (2)
+            intersection() {
+                flap();  // limit to bounds of flap
+                translate([flap_width/2, -flap_pin_width/2, 0]) {
+                    rotation = (half == 2) ? -180 : 0;  // flip upside-down for bottom
+                    gap_comp = (letter_gap_comp == true) ? -flap_gap : 0;
```

Comment:
I'm sure I'm missing something, because the renderings look good, but why is the full flap_gap offset applied to both the top and bottom half? My naive intuition is that each would get a `flap_gap/2` compensation to make up for the total `flap_gap` loss?

Reply:
Good catch. This was a bad assumption on my part - I assumed `flap_gap` was naively the gap between the flaps edges and for whatever reason you chose to store it halved. It actually refers to the gap between the edges of the flap pin _holes_ on the spool, which would otherwise be tangent.

I've refactored that variable as `flap_hole_separation` in both the main file and in `spool.scad` (6bc0e22), and created `flap_gap` as a new variable that represents the gap between the flap edges for the visible flap pair (44b1698). This new value is then halved in the letter function as expected.